homeassistant/components/snapcast/media_player.py

```diff
@@ -275,6 +279,70 @@ async def async_unjoin(self) -> None:
         """Handle the unjoin service."""
         raise NotImplementedError
 
+    @property
+    def metadata(self) -> Mapping[str, str | list[str | None]]:
+        """Get metadata from the current stream."""
+        if metadata := self.coordinator.server.stream(
+            self._current_group.stream
+        ).metadata:
+            return metadata
+
+        # Fallback to an empty dict
+        return {}
+
+    @property
+    def media_title(self) -> str | None:
+        """Title of current playing media."""
+        return cast(str, self.metadata.get("title"))
```

Comment:
One concern that I have is that we use a `.get()` to the metadata, so the result can be `Any | None`. You added the `cast` to go from `Any` to `str`. But because it's wrapped around the `.get`, it will try to cast the `None` to `str` and I think that is incorrect.

Reply:
👍  Understood. Once I figure out how to properly mock the network connection I think it should be straight forward to add proper tests for these values.